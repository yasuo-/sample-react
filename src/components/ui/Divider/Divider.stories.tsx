import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Divider } from './Divider';

export default {
  title: 'Components/UI/Divider',
  component: Divider,
  parameters: {
    layout: 'fullscreen',
  },
} as ComponentMeta<typeof Divider>;

const Template: ComponentStory<typeof Divider> = (args) => (
  <div style={{ width: '100px', height: '5px', padding: '4px' }}>
    <p style={{ margin: '8px' }}>text</p>
    <Divider {...args} />
    <p style={{ margin: '8px' }}>text2</p>
  </div>
);

const VerticalTemplate: ComponentStory<typeof Divider> = (args) => (
  <div style={{ display: 'flex', width: '200px', height: '100px', padding: '4px' }}>
    <p style={{ margin: '8px' }}>text</p>
    <Divider {...args} />
    <p style={{ margin: '8px' }}>text2</p>
  </div>
);

export const Default = Template.bind({});
export const VerticalDivider = VerticalTemplate.bind({});
export const PrimaryColorAvatar = Template.bind({});
export const SecondaryColorAvatar = Template.bind({});

Default.args = {
  orientation: 'horizontal',
  color: 'black',
};

VerticalDivider.args = {
  orientation: 'vertical',
  color: 'black',
};

PrimaryColorAvatar.args = {
  orientation: 'horizontal',
  color: 'primary',
};

SecondaryColorAvatar.args = {
  orientation: 'horizontal',
  color: 'secondary',
};
