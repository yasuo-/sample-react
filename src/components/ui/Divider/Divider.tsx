import React from 'react';

import * as Separator from '@radix-ui/react-separator';

import clsx from 'clsx';

import styles from './divider.module.css';

import type { SeparatorProps } from '@radix-ui/react-separator';

interface DividerProps extends SeparatorProps {
  color?: 'primary' | 'secondary' | 'white' | 'black';
}

/**
 * Divider
 * @param DividerProps
 * @example
 * ```
 * <Divider orientation={'horizontal'} />
 *
 * <Divider orientation={'vertical'} />
 * ```
 * @returns <Divider />
 */
export const Divider: React.FC<DividerProps> = ({
  orientation = 'horizontal',
  color = 'black',
}) => {
  const orientationClass = styles[orientation];
  const colorClass = styles[color];
  return (
    <Separator.Root orientation={orientation} className={clsx(orientationClass, colorClass)} />
  );
};

Divider.displayName = 'Divider';
