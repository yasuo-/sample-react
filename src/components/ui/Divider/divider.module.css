@tailwind components;

@layer components {
  .horizontal {
    @apply w-full h-px;
  }

  .vertical {
    @apply w-px h-full;
  }

  .primary {
    @apply bg-indigo-500;
  }

  .secondary {
    @apply bg-gray-200;
  }

  .white {
    @apply bg-white;
  }

  .black {
    @apply bg-black;
  }
}
