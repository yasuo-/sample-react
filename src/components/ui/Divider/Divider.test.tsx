import React from 'react';

import { render, screen } from '@testing-library/react';

import { Divider } from './Divider';

describe('Divider Ui Components', () => {
  it('displayName is Divider', () => {
    expect(Divider.displayName).toBe('Divider');
  });

  it('should snapshot', () => {
    const { asFragment } = render(
      <div style={{ width: '100px', height: '5px', padding: '4px' }}>
        <p style={{ margin: '8px' }}>text</p>
        <Divider />
        <p style={{ margin: '8px' }}>text2</p>
      </div>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have horizontal class name on the Divider', async () => {
    render(
      <div>
        <Divider />
      </div>
    );

    const divider = screen.queryByRole(/separator/i);
    expect(divider).toHaveClass('horizontal');
  });

  it('should have vertical class name on the Divider', async () => {
    render(
      <div style={{ width: '100px', height: '5px', padding: '4px' }}>
        <Divider orientation={'vertical'} />
      </div>
    );

    const divider = screen.queryByRole(/separator/i);
    expect(divider).toHaveClass('vertical');
  });
});
