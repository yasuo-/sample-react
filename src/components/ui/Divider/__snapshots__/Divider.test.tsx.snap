// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Ui Components should snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 100px; height: 5px; padding: 4px;"
  >
    <p
      style="margin: 8px;"
    >
      text
    </p>
    <hr
      aria-orientation="horizontal"
      class="amplify-divider amplify-divider--horizontal horizontal black"
    />
    <p
      style="margin: 8px;"
    >
      text2
    </p>
  </div>
</DocumentFragment>
`;
